src/Shared/AssemblyUtilities.cs

```diff
@@ -39,5 +45,37 @@ public static Type GetTypeInfo(this Type t)
             return t;
         }
 #endif
+
+        public static AssemblyName CloneIfPossible(this AssemblyName assemblyNameToClone)
+        {
+#if FEATURE_ASSEMBLYNAME_CLONE
+            return (AssemblyName) assemblyNameToClone.Clone();
+#else
+
+            Initialize();
+
+            if (s_assemblyNameCloneMethod == null)
+            {
+                return new AssemblyName(assemblyNameToClone.FullName);
+            }
+
+            // Try to Invoke the Clone method via reflection. If the method exists (it will on .NET
+            // Core 2.0 or later) use that result, otherwise use new AssemblyName(FullName).
+            return (AssemblyName) s_assemblyNameCloneMethod.Invoke(assemblyNameToClone, null) ??
+                   new AssemblyName(assemblyNameToClone.FullName);
+#endif
+        }
+
+        /// <summary>
+        /// Initialize static fields. Doesn't need to be thread safe.
+        /// </summary>
+        private static void Initialize()
+        {
+            if (s_initialized) return;
```

Comment:
I guess it doesn't really matter if some things race to plunk the same value down